shared/messagestore/src/main/kotlin/company/thewebhook/messagestore/consumer/KafkaConsumerImpl.kt

```diff
@@ -63,7 +63,7 @@ class KafkaConsumerImpl<T>(
             withContext(Dispatchers.IO) {
                 val records = consumer.poll(readTimeoutJavaClass)
                 records
-                    .map { Record(it.topic(), it.value()) }
+                    .map { Record(it.topic(), it.value(), "") }
```

Comment:
In Kafka, offset + partition + topic gives us a unique message id. We can use that.

I wrote an example here: https://pl.kotl.in/FzkaG2HqG

I am Base64 encoding it here so that it is a safe string (else we will get characters like �).

`resEncoded` in that example would be the 'message id'.

I wrote the decoding logic just to test it out.

Reply:
Done